pkg/cloudprovider/instance.go

```diff
@@ -381,10 +387,23 @@ func (p *InstanceProvider) getCapacityType(nodeRequest *cloudprovider.NodeReques
 	return v1alpha5.CapacityTypeOnDemand
 }
 
-// prioritizeInstanceTypes is used to eliminate less desirable instance types (like GPUs) from the list of possible instance types when
+func orderInstanceTypesByPrice(instanceTypes []*cloudprovider.InstanceType, requirements scheduling.Requirements) []*cloudprovider.InstanceType {
+	// Order instance types so that we get the cheapest instance types of the available offerings
+	sort.Slice(instanceTypes, func(i, j int) bool {
+		iPrice := instanceTypes[i].Offerings.Available().Requirements(requirements).Cheapest().Price
```

Comment:
We're not guaranteed to have any available offerings for a given instance type, right?

Reply:
Good catch!

Reply:
Updated so now we filter out any instance types that don't have available offerings that match our requirements